processor/processor.go

```diff
@@ -539,33 +539,44 @@ func getBackendEnabledDestinationTypes(writeKey string) map[string]backendconfig
 	return enabledDestinationTypes
 }
 
-func getTimestampFromEvent(event types.SingularEventT, field string) time.Time {
-	var timestamp time.Time
-	var err error
-	if _, ok := event[field]; ok {
-		timestampStr, typecasted := event[field].(string)
-		if typecasted {
-			timestamp, err = dateparse.ParseAny(timestampStr)
-		}
-		if !typecasted || err != nil {
-			timestamp = time.Now()
-		}
-	} else {
+func getTimestampFromEvent(event types.SingularEventT, field string) (timestamp time.Time) {
+	var ok bool
+	if timestamp, ok = getParsedTimestamp(event[field]); !ok {
 		timestamp = time.Now()
 	}
-	return timestamp
+	return
+}
+
+func getParsedTimestamp(input interface{}) (parsedTimestamp time.Time, valid bool) {
+	if timestampStr, typecasted := input.(string); typecasted {
+		var err error
+		parsedTimestamp, err = dateparse.ParseAny(timestampStr)
+		if err == nil {
+			valid = true
+		}
+	}
+	return
```

Comment:
Let's be explicit about returning values.
`return parsedTimestamp`

Reply:
done